app/script/conversation/ConversationCellState.js

```diff
@@ -96,14 +96,14 @@ z.conversation.ConversationCellState = (() => {
       const lastAlertMessage = conversationEntity.unreadEvents().find(_isAlert);
 
       if (lastAlertMessage) {
-        if (lastAlertMessage.is_ping()) {
-          return z.conversation.ConversationStatusIcon.UNREAD_PING;
-        }
-
         const isMissedCall = lastAlertMessage.is_call() && lastAlertMessage.was_missed();
         if (isMissedCall) {
           return z.conversation.ConversationStatusIcon.MISSED_CALL;
         }
+
+        if (lastAlertMessage.is_ping()) {
+          return z.conversation.ConversationStatusIcon.UNREAD_PING;
+        }
```

Comment:
Why without behaviour change? If there was a ping and a missed call the function will now return and display the missed call icon instead of displaying the ping icon.

Reply:
No, it will not.

`const lastAlertMessage = conversationEntity.unreadEvents().find(_isAlert);` will get the last unread message that is of any type of alert. The code you changed is essentialy a switch case deciding what type of message is connected to that.

Reply:
What you need to do instead: 

````
const hasMissedCall = conversationEntity.unreadEvents().some(messageEntity => messageEntity .is_call() && messageEntity .was_missed());
if (hasMissedCall) {
  return z.conversation.ConversationStatusIcon.MISSED_CALL;
}

const hasPing = ...
````

Reply:
Alright, thanks. Need to get used to this project... :see_no_evil: